### ISSUE TITLE
prototxt for CASIA Webface

### ISSUE THREAD

**Issue by menamet31** [wy1iu/LargeMargin_Softmax_Loss#5]
Many thanks for your work.

Could you please share train-test prototxt for CASIA experiment/params that is related to LargeMarginInnerProduct (largemargin_inner_product_param) for this "complex" case?

**Comment by wy1iu** [comment]
I will be release it along with my new CVPR 17 work "SphereFace". The SphereFace utilizes an improved L-Softmax loss and achieve the current state-of-the-art face verification performance. Please stay tuned.

**Comment by jiangxuehan** [comment]
@wy1iu : 
Could you please send your "SphereFace" paper to me (nicehjs@163.com) so i can try to implement  by myself? Thanks.

**Comment by wy1iu** [comment]
@jiangxuehan I am putting it on arXiv. You may be able to see it in a couple of days.  :)

**Comment by menamet31** [comment]
@wy1iu any news with arXiv? thanks

**Comment by wy1iu** [comment]
http://wyliu.com/papers/LiuCVPR17.pdf  Thanks.

**Comment by menamet31** [comment]
test

**Comment by menamet31** [comment]
Thank you very much for the paper. 

Am i right that LargeMarginInnerProduct layer is used in this paper? So angular softmax is the same as Large Margin softmax? Could you please release the caffe prototxt and solver for this?

**Comment by wy1iu** [comment]
No, they are not the same. The code will be different, but it should not be difficult to implement angular softmax based on L-softmax.

**Comment by xiaoboCASIA** [comment]
Thanks for your work! Are A-Softmax and Modified Softmax the special cases of L-softmax loss? It seems that when setting ||W||=1 in L-Softmax loss, the formulation is the same as A-Softmax.

**Comment by luoyetx** [comment]
@xiaoboCASIA I think you're right. It seems we can simply add a normalization layer for W before feeding it to L-Softmax Layer.

**Comment by wy1iu** [comment]
Although I did not do it like this, it is possible to implement it in the way you mentioned. @xiaoboCASIA @luoyetx  Also, a little tip for your training: You could consider to first train a model with only softmax loss on CASIA, and then train A-Softmax based on the pretrained model (still on CASIA). I train the model from scratch, but using a pretrained model is proven to be useful in my experiments. It can make lambda decay less crucial in some senses.

**Comment by wy1iu** [comment]
Yes, it can be viewed as special cases. It shows much better performance in face recognition. @xiaoboCASIA

**Comment by Chetter2** [comment]
@wy1iu do you still plan to release Angular-Softmax?
